devicetypes/zooz/zooz-zse11-q-sensor.src/zooz-zse11-q-sensor.groovy

```diff
@@ -0,0 +1,448 @@
+		logForceWakeupMessage("The sensor values will be requested the next time the device wakes up.")
+	} else {
+		sendHubCommand(getRefreshCmds())
+	}
+}
+
+void logForceWakeupMessage(msg) {
+	log.debug "${msg}  You can force the device to wake up immediately by holding the z-button for 3 seconds."
+}
+
+List<String> getRefreshCmds() {
+	return [
+		secureCmd(zwave.sensorBinaryV2.sensorBinaryGet(sensorType: motionSensorType)),		
+		sensorMultilevelGetCmd(tempSensorType),
+		sensorMultilevelGetCmd(lightSensorType),
+		sensorMultilevelGetCmd(humiditySensorType),
+		batteryGetCmd()
+	]
+}
+
+String sensorMultilevelGetCmd(sensorType) {
+	def scale = (sensorType == tempSensorType ? 0 : 1)
+	return secureCmd(zwave.sensorMultilevelV5.sensorMultilevelGet(scale: scale, sensorType: sensorType))
+}
+
+String batteryGetCmd() {
+	return secureCmd(zwave.batteryV1.batteryGet())
+}
+
+String secureCmd(cmd) {
+	if (zwaveInfo?.zw?.contains("s")) {
+		return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		return cmd.format()
+	}
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description, commandClassVersions)
+	if (cmd) {
+		zwaveEvent(cmd)
+	} else {
+		log.warn "Unable to parse: $description"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCmd = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCmd) {
+		zwaveEvent(encapsulatedCmd)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	log.debug "Device Woke Up"
+	List<String> cmds = []
+
+	if (state.pendingRefresh) {
+		state.pendingRefresh = false
+		cmds += getRefreshCmds()
+	}
+
+	cmds += getConfigCmds()
+	
+	if (!cmds) {
+		cmds << batteryGetCmd()
+	}
+
+	cmds << secureCmd(zwave.wakeUpV1.wakeUpNoMoreInformation())
+	sendHubCommand(cmds)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	int val = (cmd.batteryLevel == 0xFF ? 1 : cmd.batteryLevel)
+	if (val > 100) {
+		val = 100
+	} else if (val < 1) {
+		val = 1
+	}
+
+	if (device.currentValue("powerSource") != "battery") {
+		log.debug "powerSource is battery"
+		sendEvent(name:"powerSource", value:"battery")
+	}
+
+	log.debug "battery is ${val}%"
+	sendEvent(name:"battery", value:val, unit:"%", isStateChange:true)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	switch (cmd.sensorType) {
+		case tempSensorType:
+			def temp = convertTemperatureIfNeeded(cmd.scaledSensorValue, (cmd.scale ? "F" : "C"), cmd.precision)
+			sendTempEvent(temp)
+			break
+		case lightSensorType:
+			sendLightEvent(cmd.scaledSensorValue)
+			break
+		case humiditySensorType:
+			sendHumidityEvent(cmd.scaledSensorValue)
+			break
+		default:
+			log.debug "Unhandled: ${cmd}"
+	}
+}
+
+void sendTempEvent(reportedVal, boolean onlyIfNew=false) {
```

Comment:
After the same event value is sent then the previously created event is updated and that's valuable too.
I wouldn't use flag: `onlyIfNew` (here and in other functions).
@greens any thoughts on this ?